addons/mail/static/src/components/message/message.xml

```diff
@@ -3,46 +3,44 @@
                     'o-not-discussion': !messageView.message.isDiscussionOrNotification,
                     'o-notification': messageView.message.message_type === 'notification',
                     'o-selected': messageView.isSelected,
-                    'o-squashed': messageView.isSquashed,
+                    'o-squashed pt-1': messageView.isSquashed,
                     'o-starred': messageView.message.isStarred,
-                    'mt-3': !messageView.isSquashed and messageView.messageListViewMessageViewItemOwner,
-                    'bg-white': messageView.message.isDiscussionOrNotification and !messageView.isSelected and !(messageView.message.isHighlighted or messageView.isHighlighted) and !messageView.isActive,
-                    'bg-100': (!messageView.message.isDiscussionOrNotification and !messageView.isActive and !messageView.isSelected) or (messageView.isActive and !messageView.isSelected and messageView.message.isDiscussionOrNotification and !(messageView.message.isHighlighted or messageView.isHighlighted)),
-                    'bg-200': !messageView.message.isDiscussionOrNotification and messageView.isActive and !messageView.isSelected,
-                    'bg-400': messageView.isSelected and !(messageView.message.isHighlighted or messageView.isHighlighted),
+                    'o-currentAuthor': messageView.message.isCurrentUserOrGuestAuthor,
+                    'mt-1': !messageView.isSquashed,
+                    'mt-5': !messageView.isSquashed and messageView.messageListViewOwner,
+                    'px-3': !messageView.isInChatWindow,
+                    'px-1': messageView.isInChatWindow,
                     'opacity-50': (messageView.messageListViewMessageViewItemOwner and messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.replyingToMessageView) and !messageView.isSelected,
                 }" t-attf-class="{{ messageView.extraClass }} {{ className }}" role="group" t-att-aria-label="messageView.message.messageTypeText" t-on-click="messageView.onClick" t-on-mouseenter="messageView.onMouseenter" t-on-mouseleave="messageView.onMouseleave" t-att-data-id="messageView.message.id" t-att-data-message-local-id="messageView.message.localId"
                 t-ref="root"
             >
-                <MessageInReplyToView t-if="messageView.messageInReplyToView" record="messageView.messageInReplyToView"/>
-                <div t-if="messageView.isActive and messageView.messageActionList" class="o_Message_actionListContainer position-absolute top-0 end-0 mt-n3 px-3" t-att-class="{ 'o-squashed': messageView.isSquashed }">
-                    <MessageActionList record="messageView.messageActionList"/>
+                <div class="o_Message_inReplyToWrap" t-att-class="{ 'me-5': messageView.isInDiscuss or messageView.isInChatWindowAndIsAlignedLeft, 'ms-5': messageView.isInChatWindowAndIsAlignedRight, 'd-flex justify-content-end': messageView.isInChatWindowAndIsAlignedRight }">
```

Comment:
can group `ms-5` and `d-flex justify-content-end` together (they have exact same condition)